src/static/js/common_widgets.js

```diff
@@ -906,8 +917,26 @@ $.widget("asm.createpayment", {
         $("#pm-vatamount").currency("value", o.vatamount || 0);
         $("#pm-comments").html( o.comments );
         $("#pm-due").date("today");
-        $("#dialog-payment .paymentsalestax").toggle(o.vat);
+        //$("#dialog-payment .paymentsalestax").toggle(o.vat); # Took this out as it was overiding new behaviour
+        $("#pm-vat").change();
+        let vatrates = [];
+        let defaulttaxrateval = "";
+        $.each(controller.taxrates, function(taxratecount, taxrate) {
+            let optval = taxrate.ID + "|" + taxrate.TAXRATE;
+            if (taxrate.ID == config.integer("AFDefaultTaxRate")) {
+                defaulttaxrateval = optval;
+            }
+            vatrates.push({ID: optval, TAXRATENAME: taxrate.TAXRATENAME});
+        });
+        let vatrateoptions = html.list_to_options(vatrates, "ID", "TAXRATENAME" );
```

Comment:
This loop and the two variables vatrates and defaulttaxrateeval seems to be superfluous to me?  Think it could be written as:

```
$("#pm-vatratechoice").html( html.list_to_options(controller.taxrates, "ID", "TAXRATENAME") );
$("#pm-vatratechoice").val(config.integer("AFDefaultTaxRate"));
```